tests/py/dynamo/conversion/test_group_norm_aten.py

```diff
@@ -6,155 +6,56 @@
             def forward(self, x):
                 return torch.ops.aten.native_group_norm.default(
-                    x,
-                    torch.ones((6,)),
-                    torch.zeros((6,)),
-                    3,
-                    6,
-                    224,
-                    2,
-                    1e-05,
+                    x, None, None, 3, 6, 224, 2, 1e-05
                 )[0]
 
         inputs = [torch.randn(3, 6, 224)]
-        self.run_test(
-            GroupNorm(),
-            inputs,
-        )
+        self.run_test(GroupNorm(), inputs, use_dynamo_tracer=True)
 
-    def test_groupnorm2d(self):
+    def test_groupnorm_2d(self):
         class GroupNorm(torch.nn.Module):
-            def forward(self, x):
+            def forward(self, x, weight, bias):
                 return torch.ops.aten.native_group_norm.default(
-                    x,
-                    torch.ones((6,)),
-                    torch.zeros((6,)),
-                    3,
-                    6,
-                    224 * 224,
-                    2,
-                    1e-05,
+                    x, weight, bias, 3, 6, 224 * 224, 2, 1e-05
                 )[0]
 
-        inputs = [torch.randn(3, 6, 224, 224)]
-        with torch.no_grad():
-            self.run_test(
-                GroupNorm(),
-                inputs,
-            )
+        inputs = [torch.randn(3, 6, 224, 224), torch.ones(6), torch.zeros(6)]
+        self.run_test(GroupNorm(), inputs, use_dynamo_tracer=True)
 
     def test_groupnorm_sd(self):
         class GroupNorm(torch.nn.Module):
-            def forward(self, x):
+            def forward(self, x, weight, bias):
                 return torch.ops.aten.native_group_norm.default(
-                    x,
-                    torch.randn((320,)).half(),
-                    torch.randn((320,)).half(),
-                    2,
-                    320,
-                    4096,
-                    32,
-                    1e-05,
+                    x, weight, bias, 2, 320, 64 * 64, 32, 1e-05
                 )[0]
 
-        inputs = [torch.randn(2, 320, 64, 64).half()]
-        with torch.no_grad():
-            self.run_test(
-                GroupNorm(),
-                inputs,
-            )
+        inputs = [
+            torch.randn(2, 320, 64, 64, dtype=torch.half),
+            torch.randn(320, dtype=torch.half),
+            torch.randn(320, dtype=torch.half),
+        ]
+        self.run_test(GroupNorm(), inputs, precision=torch.half, use_dynamo_tracer=True)
 
     @parameterized.expand(
         [
             (5, 4, 4, 2, (2, 4, 2), (3, 4, 2), (5, 4, 4)),
             (5, 4, 2 * 2, 2, (2, 4, 2, 2), (3, 4, 2, 2), (5, 4, 2, 2)),
             (5, 9, 6 * 3, 3, (3, 9, 3, 3), (4, 9, 3, 3), (5, 9, 6, 3)),
-            (8, 9, 6 * 6, 3, (3, 9, 2, 3, 2), (5, 9, 3, 3, 2), (8, 9, 6, 3, 2)),
+            (8, 9, 6 * 3 * 2, 3, (3, 9, 2, 3, 2), (5, 9, 3, 3, 2), (8, 9, 6, 3, 2)),
         ]
```

Comment:
So in these dynamic cases the input is  (8, 9, 6, 3, 2). Say suppose it gets the input as (5, 9, 3, 3, 2) is that a valid input? But then in the code we do not use H*W anywhere explicitly so the code I am assuming would not error out in that case too

Reply:
Since `run_test_with_dynamic_shape` uses `max_shape` for actual testing, the `N, C, HxW` arguments are intentionally specified using values from `max_shape` for expected behavior. In practice PyTorch will provide those values for users of `nn.GroupNorm`. And as you have mentioned, `HxW` isn't used in the code anyway since we let TRT do the group division and flattening.

Reply:
ok so nn.GroupNorm would take in the values from the input shape, but then as @zewenli98 pointed `torch.ops.aten.native_group_norm.default` would error out. Two points
1. If you introduce a check for H*W, then some dynamic cases for min_shape and opt_shape would be invalid
2. But then since `run_test_with_dynamic_shape` will just check with `max_shape`, so tests will not fail. But then should we specify min_shape and opt_shape for cases not valid for torch.ops.aten.native_group_norm.default (but then valid for nn.GroupNorm). 
@HolyWu  and @zewenli98 what do you guys think?

Reply:
As @HolyWu said, 

> N, C, HxW arguments are calculated and supplied by PyTorch for the users of torch.nn.GroupNorm depending on the shape of input tensor. It's only us that directly call the aten operator rather than using torch.nn.GroupNorm like normal users do. 

> Actually adding check for N and HxW is easy for static-shaped input. The real problem for me is dynamic-shaped input. Since the value is -1 for dynamic dim, either impl.shape.shape or impl.shape.get_shape_with_dynamic_shape need to be invoked to get the real shape. But the returned value is of type trt.ITensor and I don't know how to compare it with scalar type.

I have two solutions in my mind:
1. Indeed, users are rare to directly call aten ops, so I think the current implementation is fine.
2. If you really want to check N or HxW, you can call `impl.elementwise.eq` and then [TRT assertion layer](https://docs.nvidia.com/deeplearning/tensorrt/api/python_api/infer/Graph/Network.html#tensorrt.INetworkDefinition.add_assertion).

Reply:
I think I'm fine with option 1 too.